fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/mixin/networking/ClientConnectionMixin.java

```diff
@@ -110,4 +120,27 @@ private void disconnectAddon(CallbackInfo ci) {
 	public Collection<Identifier> fabric_getPendingChannelsNames(NetworkPhase state) {
 		return this.playChannels.computeIfAbsent(state, (key) -> Collections.newSetFromMap(new ConcurrentHashMap<>()));
 	}
+
+	@Override
+	public void fabric_invokeCookieCallback(Identifier cookieId, byte[] cookie) {
+		CompletableFuture<byte[]> future = pendingCookieRequests.remove(cookieId);
+		if (future == null) return;
+		future.complete(cookie);
```

Comment:
Question: What thread does this complete on? I would expect the network thread during config, and the main thread during play

Reply:
I think it always runs the network thread, as its triggered by the packet response (unless packets process in the main thread in play? I didn't think they did)